File: Leetcode_Python/min-number-of-arrows.py
class Solution(object):
    def findMinArrowShots(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        
        sorted_points = sorted(points,key=lambda x:x[1])
        thrown_arrows = 1
        curr_end = sorted_points[0][1]

        for i in range(1, len(sorted_points)):
            start = sorted_points[i][0]
            end = sorted_points[i][1]

            if (start > curr_end):
                thrown_arrows += 1
                curr_end = end

        return thrown_arrows

File: Leetcode_JS/min-number-of-arrows.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShot = function(points) {
    
    arrows = 1
    sortedPoints = points.sort((a,b) => a[1]-b[1])
    let currEnd = sortedPoints[0][1]

    for (let i = 1; i < points.length; i++) {
        let start = (points[i][0])

        if (start > currEnd) {
            arrows++
            currEnd = points[i][1]
        }
    }
    return arrows
};